### DISCUSSION TITLE
factory reset when using push button

### DISCUSSION THREAD

**Discussion by ALfuhrmann** [arendst/Tasmota#14910] [Q&A] [answered]
I am using a Shelly 1 with the Tasmota  11.0.0. I want to toggle the lights with push button switches already existing in my flat.

After initial configuration using the default Shelly 1 template, (switchmode default 0), the lights were on, and switched off as long as I pressed the button. I thought changing the template to button_n would do the trick, and it worked initially. Then the Tasmota did a factory reset. Panic.

So I searched and found that this is more or less expected behavior: press the button long enough for a factory reset.

I tried to configure the button as normally closed and tried "button_i" and "button_in" in the WebUI to no avail  (where are these values documented?)

So I resorted to changing the switchmode via console to "4", which finally worked: toggle, but no reset.
I am going to make a note of this, but the next time I configure a Shelly and try to copy the settings from the WebUI, I am probably going to make the same mistake again.

My question: is there no way to configure a normally closed button via the WebUI to NOT result in a factory reset?

**Comment by ascillato** [comment, marked as answer]
Hi,

The Shelly 1 template is: `{"NAME":"Shelly 1","GPIO":[1,1,0,1,224,192,0,0,0,0,0,0,0,0],"FLAG":0,"BASE":46}` and by default that expects the wiring to be as the Standard Shelly specification for Shelly 1 (that is a common switch that is NO: Normal Open). If your input is different, you should NOT change the template. You should configure the behavior of the shelly input by `switchmode` command and by setoptions.

> I tried to configure the button as normally closed and tried "button_i" and "button_in" in the WebUI to no avail (where are these values documented?)

At the Tasmota docs: https://tasmota.github.io/docs/Components/#gpios (if you use the search bar on the top of the docs, you can find this page)

> My question: is there no way to configure a normally closed button via the WebUI to NOT result in a factory reset?

- Use the default Shelly 1 template `{"NAME":"Shelly 1","GPIO":[1,1,0,1,224,192,0,0,0,0,0,0,0,0],"FLAG":0,"BASE":46}`.
The input must be `Switch_n 1` due to it is required for Shelly hardware to work properly. Any other option will produce unexpected results due to incorrect hardware management.
- Set your input as a inverted pushbutton. Go to the console and type `SWITCHMODE1 4`.
This option is what you need for changing the default to your use-case (an inverted push button instead of a N.O. Switch).

If you want to disable the factory reset feature, use `SetOption1 1`. More info at https://tasmota.github.io/docs/Commands/#setoptions, https://tasmota.github.io/docs/Buttons-and-Switches/#multi-press-functions and old issues.

**Reply by ALfuhrmann** [reply]
Thank you!

**Comment by fabian727** [comment]
https://tasmota.github.io/docs/Buttons-and-Switches/#button
holds a kind of "schematic"
https://tasmota.github.io/docs/Buttons-schematics/

I load this picture every time I am working with buttons/ switches.
